### ISSUE TITLE
Compiler does not terminate when called with undefined phase

### ISSUE THREAD

**Issue by dmj** [Schematron/schematron#87]
If the Schematron compiler is called with a phase and this phase is not defined, it reports an error via xml:message but still compiles the validation stylesheet. 

I was thinking about a use case where this behavior is neccessary or helpful to no avail.

**Comment by dmj** [comment]
If I follow the 2006 spec closely then the behavior of the Skeleton is clearly wrong. In section 6.3 (Schema Semantics) it requires the active phase to be a member of the schema.

The note on 6.3 reads:

> In  natural  language,  a  document  is  valid  against  a  schema  in  a  phase  if:  
There  exists  an  instance,  schema  and active-phase combination where, for each context, pattern, rule and assertion (the context being a member of that instance, the active phase being a member of the schema, the pattern being a member of that active phase, the rule being a member of that pattern, the assertion being a member of that rule), the following is true: if the context of an instance matches the rule, and that context has not been matched by a previous rule in the same pattern, then the particular assertion evaluates to true when evaluated with the particular context and instance.

**Comment by tgraham-antenna** [comment]
Yes, but if we're quoting the spec, Clause 6.1 of the 2016 spec says:

> A Schematron validator is a function over the following:
> ...
> a phase name, or #ALL if all patterns shall be active patterns, or #DEFAULT if the defaultPhase
attribute on the schema element shall be used;

So, in 2016, the consequence of an undefined phase name was just an empty function.

On the face of it, an undefined phase name should just terminate the compilation, but I'd have to look to see if it's currently possible to later run the compiled stylesheet starting from a different phase.

**Comment by dmj** [comment]
> So, in 2016, the consequence of an undefined phase name was just an empty function.

Ok, missed this part.

> On the face of it, an undefined phase name should just terminate the compilation, but I'd have to look to see if it's currently possible to later run the compiled stylesheet starting from a different phase.

It isn't. The Skeleton (as well as SchXslt) compiles a validation stylesheet for a specific phase.
